src/main/java/com/cleanroommc/groovyscript/compat/loot/GroovyLootFunction.java

```diff
@@ -25,7 +25,10 @@ public GroovyLootFunction(LootCondition[] conditions, Closure<Object> function)
         if (Arrays.equals(function.getParameterTypes(), new Class[]{ItemStack.class, Random.class, LootContext.class})) {
             this.function = function;
         } else {
-            GroovyLog.msg("Error creating custom loot function: function must take the following parameters (net.minecraft.item.ItemStack, java.util.Random, net.minecraft.world.storage.loot.LootContext).").error().post();
+            GroovyLog.msg("Error creating LootFunction:")
+                    .add("function must take the following parameters (net.minecraft.item.ItemStack, java.util.Random, net.minecraft.world.storage.loot.LootContext)")
+                    .error()
+                    .post();
             this.function = null;
```

Comment:
Enforcing parameter types i something i dont really like. In groovy you can normally just omit the types.

Reply:
Interesting, I understand why someone would want to omit the types for brevity but I also just figured a very clear error would help save us headaches from end users asking in the discord "why am I getting 'java.util.Random does not have method .setCount(int)' error"

I can revert this and change the docs if this is the consensus however.

Reply:
set warnings about types to debug level and made type specification optional: [905b168](https://github.com/CleanroomMC/GroovyScript/pull/95/commits/905b16810d23ca19426a8647bd3ad2112600e09d)